### ISSUE TITLE
Suppress warnings with sklearn API when run from terminal

### ISSUE THREAD

**Issue by Jakobhenningjensen** [microsoft/LightGBM#3643]
## How you are using LightGBM?

LightGBM component:  LightgbmClassifier

## Environment info

Operating System:

CPU/GPU model: Intel i7-8550 1.8 GHz  / NA

C++ compiler version: 

CMake version:

Java version:

Python version: 3.7.9

R version:

Other:

LightGBM version or commit hash: v. 3.0.0

## Error message and / or logs

[LightGBM] [Warning] Accuracy may be bad since you didn't set num_leaves and 2^max_depth > num_leaves

## Steps to reproduce
1. Create python file `main.py`
```python
from lightgbm import LGBMClassifier as LGBM
from sklearn.datasets import make_classification
from sklearn.model_selection import train_test_split
from sklearn.metrics import roc_auc_score
import optuna
optuna.logging.set_verbosity(optuna.logging.ERROR)

#Create dataset
print("Creating dataset")
data = make_classification(10000)
X = data[0]
y= data[1]

X_train,X_val,y_train,y_val = train_test_split(X,y,test_size=0.1)

print("Training model (incl. verbose=-1)")
model = LGBM(objective="binary",max_depth=2^32,verbose=-1)
model.fit(X_train,y_train)
pred = model.predict_proba(X_val)
auc = roc_auc_score(y_val,pred[:,1])
print(f"Val AUC: {auc}")

```
2.a) Create a bash-script which runs the script `main.py`  (using a virtual environment `vir_env`) and prints issues to `log.txt`
```bash
#!/bin/bash/
python(){
vir_env\\Scripts\\python.exe "$@"
}

pyton -u main.py>>log.txt 2>&1
```

or
2.b) Run `python main.py` in a terminal

and
2.c) Open VScode and run the file in the interactive window

3.a) `log.txt` now contains the following

```
Creating dataset
Defining Model (incl. verbose=-1)
[LightGBM] [Warning] Accuracy may be bad since you didn't set num_leaves and 2^max_depth > num_leaves
Val AUC: 0.9462085573476702
```

3.b) The terminal prints the same as in 3.a

3.c) There are no warnings in the VScode interactive window, only the print-statements (as expected)

**Comment by jameslamb** [comment]
Thanks for using LightGBM, and for the thorough report!

I believe this is a duplicate of https://github.com/microsoft/LightGBM/issues/3641. See https://github.com/microsoft/LightGBM/issues/3641#issuecomment-742939002 for the root cause of this specific warning not respecting your setting for `verbose`. We'd welcome a pull request to fix it, if you have the time and interest!

If I've misunderstood, please leave a comment explaining how this is different from #3641 and we can re-open it.
